### PR TITLE
Set authorised volume for 2pt

### PR DESCRIPTION
**Pull request by Beckyrose200**
https://eaflood.atlassian.net/browse/WATER-4440

As part of the ongoing work for the two-part tariff review screens, this PR introduces the ability for the users to change the authorised volume on the charge reference.

### CODE REVIEW

File: app/views/bill-runs/amend-authorised-volume.njk [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            text: "ML"
+          }
+        }) }}
+
+        {# Hidden input for authorised volume #}
+        <input type="hidden" name="totalBillableReturns" value="{{ chargeReference.totalBillableReturns }}">
+        <input type="hidden" name="minVolume" value="{{ chargeCategory.minVolume }}">
+        <input type="hidden" name="maxVolume" value="{{ chargeCategory.maxVolume }}">
+
+        {{ govukButton({ text: 'Confirm' }) }}
```
Comment (Jozzey) [inline app/views/bill-runs/amend-authorised-volume.njk:73]: I think the norm is to add `preventDoubleClick: true` to the button

File: app/views/bill-runs/charge-reference-details.njk [hunk 2]
```diff
@@ -66,7 +66,8 @@
       <span class="inline govuk-!-margin-right-3">
         {{ govukButton({
           text: "Change the authorised volume",
-          classes: "govuk-button--secondary"
+          classes: "govuk-button--secondary",
+          href: chargeReference.id + '/amend-authorised-volume'
```
Comment (Jozzey) [inline app/views/bill-runs/charge-reference-details.njk:70]: I think the norm is to add `preventDoubleClick: true` to the button

File: test/presenters/bill-runs/two-part-tariff/amend-authorised-volume.presenter.test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        })
+      })
+    })
+  })
+})
+
+function _billRun () {
+  return {
+    id: '6620135b-0ecf-4fd4-924e-371f950c0526',
+    fromFinancialYearEnding: 2023,
```
Comment (Jozzey) [inline test/presenters/bill-runs/two-part-tariff/amend-authorised-volume.presenter.test.js:78]: Don't think you need this `fromFinancialYearEnding: 2023,`

**Review by Jozzey — CHANGES_REQUESTED** [review]
Just a few little things

File: app/services/bill-runs/two-part-tariff/fetch-authorised-volume.service.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  return { billRun, reviewChargeReference }
+}
+
+async function _fetchBillRun (billRunId) {
+  return BillRunModel.query()
+    .findById(billRunId)
+    .select(
+      'id',
+      'fromFinancialYearEnding',
```
[the remark this answers is not in the document]
Comment (Beckyrose200) [reply app/services/bill-runs/two-part-tariff/fetch-authorised-volume.service.js]: Good spot, thank you

File: app/views/bill-runs/charge-reference-details.njk (hunk 2 above)
Comment (Beckyrose200) [reply app/views/bill-runs/charge-reference-details.njk:71]: Good spot, thank you. Ive missed this out in a few places. I have created a [new PR](https://github.com/DEFRA/water-abstraction-system/pull/1050) to make this consistent.

**Review by Jozzey — APPROVED** [review]

**Pull request merged**